torch/cuda/_utils.py

```diff
@@ -1,11 +1,10 @@
 import torch
-from typing import Union
-from torch.types import Device
+from typing import Any
 # The _get_device_index has been moved to torch.utils._get_device_index
 from torch._utils import _get_device_index as _torch_get_device_index
 
 
-def _get_device_index(device: Union[Device, str, int, None], optional: bool = False,
+def _get_device_index(device: Any, optional: bool = False,
```

Comment:
So you said:

> Union types have been replaced with Any and then type refined in the function body.

But I don't understand why you need to do this; from the perspective of an eager mode user at all, Any is worse (though the situation here is less bad because this is an internal helper). Is this because JIT doesn't support unions?

Reply:
Yes, it's because JIT doesn't support unions. 😢